packages/components/src/flyout/flyout/component.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { PopoverDisclosure, Portal } from 'reakit';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo, cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { contextConnect } from '../../ui/context';
+import { FlyoutContext } from '../context';
+import { useFlyoutResizeUpdater } from '../utils';
+import FlyoutContent from '../flyout-content';
+import { useUpdateEffect } from '../../utils/hooks';
+import { useFlyout } from './hook';
+
+/**
+ *
+ * @param {import('../../ui/context').PolymorphicComponentProps<import('../types').Props, 'div'>} props
+ * @param {import('react').Ref<any>}                                                              forwardedRef
+ */
+function Flyout( props, forwardedRef ) {
+	const {
+		children,
+		elevation,
+		label,
+		maxWidth,
+		onVisibleChange,
+		trigger,
+		popover,
+		...otherProps
+	} = useFlyout( props );
+
+	const resizeListener = useFlyoutResizeUpdater( {
+		onResize: popover.unstable_update,
+	} );
+
+	const uniqueId = `flyout-${ popover.baseId }`;
+	const labelId = label || uniqueId;
+
+	const contextProps = useMemo(
+		() => ( {
+			label: labelId,
+			popover,
+		} ),
+		[ labelId, popover ]
+	);
+
+	const triggerContent = useCallback(
+		( triggerProps ) => {
+			return cloneElement( trigger, triggerProps );
+		},
+		[ trigger ]
+	);
+
+	useUpdateEffect( () => {
+		onVisibleChange?.( popover.visible );
+	}, [ popover.visible ] );
```

Comment:
Not sure if any of this could be also moved to the hook

Reply:
Not an issue anymore, since we're going to used `styled`  — which moves us away from hooks